hibernate-reactive-core/src/test/java/org/hibernate/reactive/HqlSelectNewConstructorTest.java

```diff
@@ -0,0 +1,127 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.vertx.junit5.VertxTestContext;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static jakarta.persistence.CascadeType.PERSIST;
+import static jakarta.persistence.FetchType.LAZY;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class HqlSelectNewConstructorTest extends BaseReactiveTest {
```

Comment:
Sorry, I should have mentioned this before, but we already have [a class that tests HQL queries](https://github.com/hibernate/hibernate-reactive/blob/a98716ba3b861ee1194895566c963c216e7c7421/hibernate-reactive-core/src/test/java/org/hibernate/reactive/HQLQueryTest.java#L31).

We create a SessionFactory for each separate class, and in the long run it could slow down the build having too many of them.